src/include/souffle/datastructure/Brie.h

```diff
@@ -55,7 +57,41 @@
 
 namespace souffle {
 
-namespace detail {
+template <unsigned Dim>
+class Trie;
+
+namespace detail::brie {
+
+// FIXME: These data structs should be parameterised/made agonistic `RamDomain`.
```

Comment:
I'm fairly sure you meant
```suggestion
// FIXME: These data structs should be parameterised/made agnostic `RamDomain`.
```

Reply:
Correct.

The change to `Tuple` is trivial in practice AFAIK. `Tuple` and `std::array` are both wrappers around a c-array and basically do the same thing. `std::array`, however, is more commonly known (analysis tool support, std lib bindings, etc).

Reply:
Oh, I completely support the change to `std::array` for all the reasons you mention. I even started the same changes in the past, but was blocked by code that has apparently all been removed now.